ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PeerDiscoveryController.java

```diff
@@ -483,7 +486,23 @@ RecursivePeerRefreshState getRecursivePeerRefreshState() {
    */
   private void refreshTable() {
     final Bytes target = Peer.randomId();
-    final List<DiscoveryPeer> initialPeers = peerTable.nearestBondedPeers(Peer.randomId(), 16);
+
+    final List<DiscoveryPeer> initialPeers;
+    if (includeBootnodesOnPeerRefresh) {
+      bootstrapNodes.stream()
+          .filter(p -> p.getStatus() != PeerDiscoveryStatus.BONDED)
+          .forEach(p -> p.setStatus(PeerDiscoveryStatus.KNOWN));
+
+      // If configured to retry bootnodes during peer table refresh, include them
+      // in the initial peers list.
+      initialPeers =
+          Stream.concat(
```

Comment:
we could just do a `initialPeers.addAll(bootstrapNodes)` here if we were to do line 504 before the if.